plugins/outputs/warp10/warp10.go

```diff
@@ -0,0 +1,293 @@
+package warp10
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+const (
+	defaultClientTimeout = 15 * time.Second
+)
+
+// Warp10 output plugin
+type Warp10 struct {
+	Prefix  string
+	WarpURL string
+	Token   string
+	Timeout internal.Duration `toml:"timeout"`
+	client  *http.Client
+	tls.ClientConfig
+}
+
+var sampleConfig = `
+  # prefix for metrics class Name
+  prefix = "Prefix"
+  ## POST HTTP(or HTTPS) ##
+  # Url name of the Warp 10 server
+  warp_url = "WarpUrl"
+  # Token to access your app on warp 10
+  token = "Token"
+  # Warp 10 query timeout, by default 15s
+  timeout = "15s"
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+`
+
+// MetricLine Warp 10 metrics
+type MetricLine struct {
+	Metric    string
+	Timestamp int64
+	Value     string
+	Tags      string
+}
+
+func (w *Warp10) createClient() (*http.Client, error) {
+	tlsCfg, err := w.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	if w.Timeout.Duration == 0 {
+		w.Timeout.Duration = defaultClientTimeout
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: w.Timeout.Duration,
+	}
+
+	return client, nil
+}
+
+// Connect to warp10
+func (w *Warp10) Connect() error {
+	client, err := w.createClient()
+	if err != nil {
+		return err
+	}
+
+	w.client = client
+	return nil
+}
+
+// GenWarp10Payload compute Warp 10 metrics payload
+func (w *Warp10) GenWarp10Payload(metrics []telegraf.Metric, now time.Time) string {
+	collectString := make([]string, 0)
+	for _, mm := range metrics {
+
+		for _, field := range mm.FieldList() {
+
+			metric := &MetricLine{
+				Metric:    fmt.Sprintf("%s%s", w.Prefix, mm.Name()+"."+field.Key),
+				Timestamp: now.UnixNano() / 1000,
+			}
+
+			metricValue, err := buildValue(field.Value)
+			if err != nil {
+				log.Printf("E! [outputs.warp10] Could not encode value: %v", err)
+				continue
+			}
+			metric.Value = metricValue
+
+			tagsSlice := buildTags(mm.TagList())
+			metric.Tags = strings.Join(tagsSlice, ",")
+
+			messageLine := fmt.Sprintf("%d// %s{%s} %s\n", metric.Timestamp, metric.Metric, metric.Tags, metric.Value)
+
+			collectString = append(collectString, messageLine)
+		}
+	}
+	return fmt.Sprint(strings.Join(collectString, ""))
+}
+
+// Write metrics to Warp10
+func (w *Warp10) Write(metrics []telegraf.Metric) error {
+
+	if len(metrics) == 0 {
```

Comment:
Instead of this, check if the payload is empty a few lines down and return then.